### ISSUE TITLE
Spacy Initialize Error

### ISSUE THREAD

**Issue by mmmille5** [quanteda/spacyr#83]
Hi - I am encountering issues when trying to initialize spacy. I have confirmed that I have properly downloaded the file in anaconda, but am getting the following error message:

```r
Finding a python executable with spacy installed...
spaCy (language model: en) is installed in C:\Users\xxxx\AppData\Local\Continuum\anaconda3\python.exe
Error in py_run_file_impl(file, local, convert) : 
  TypeError: 'int' object is not callable
```

Could you please advise on what I am doing wrong? I am new to python and R, but see great value in using spacyr for NLP. 

Thank You!

**Comment by kbenoit** [comment]
A few questions:
1) Have you verified that spaCy works from outside of R? 
2) Have you tried installing the newest **spacyr** version, from GitHub?

Let us know and we can try to help further.

**Comment by amatsuo** [comment]
It would be very helpful if you can report whether `reticulate` is working in your environment. 

In particular, could you try run examples in the following and let us know that it is working?
https://rstudio.github.io/reticulate/articles/introduction.html

Also, let us know whether you have more than one python in your system (you can check it by entering `where python` in your console.

**Comment by mmmille5** [comment]
Kenneth - I have verified that spaCy works outside of R. I tried installing the newest spacyr version from GitHub, but am lost on how to do that. I had previously downloaded the Cran version.

Aki - 
I tried a few of the reticulate tests and ran into the following errors. 

library(reticulate)
> os <- import("os")
> os$chdir("tests")
Error in py_call_impl(callable, dots$args, dots$keywords) : 
  FileNotFoundError: [WinError 2] The system cannot find the file specified: 'tests'

> difflib <- import("difflib")
> difflib$ndiff(foo,bar)
Error in py_resolve_dots(list(...)) : object 'foo' not found

I did confirm that I only have one instance of anaconda in my system (anaconda 3)

Sorry if I am not being much help on this - I am trying my best to learn what I need to in order to resolve the issue.

**Comment by kbenoit** [comment]
Try: 
```r
devtools::install_github("kbenoit/spacyr", build_vignettes = FALSE)
```
and then see if `spacy_inititialize()` works.

**Comment by myeomans** [comment]
Sorry to say, but I'm having a similar error... when I upgrade to SpaCy 2.0.x, the spacy_initialize function crashes RStudio. It breaks when I run it from the terminal... this is the report:

"Intel MKL FATAL ERROR: Cannot load libmkl_intel_thread.dylib"

I'm stumped! Any thoughts? Will revert back to 1.9 for now, but given how helpful spacyr has already been, I am eager to try it with the new version, too!

**Comment by kbenoit** [comment]
@myeomans what is the output of `sessionInfo()`? Be sure you are running everything up to date, esp. **reticulate**.

**Comment by myeomans** [comment]
R version 3.4.1 (2017-06-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.1

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

other attached packages:
[1] reticulate_1.3.1 spacyr_0.9.4    

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.14        grid_3.4.1         
 [3] plyr_1.8.4          jsonlite_1.5       
 [5] gtable_0.2.0        scales_0.5.0       
 [7] ggplot2_2.2.1       rlang_0.1.4        
 [9] lazyeval_0.2.1      data.table_1.10.4-3
[11] tools_3.4.1         munsell_0.4.3      
[13] yaml_2.1.14         compiler_3.4.1     
[15] colorspace_1.3-2    tibble_1.3.4

**Comment by kbenoit** [comment]
OK, looks like you are up to date. So:

1. From the Terminal (the bash Terminal, not R), does spaCy execute properly, by trying the tests at https://spacy.io/usage/#tests?

2. Have you installed Xcode, using `xcode-select --install` from the Terminal? (You will then need to launch Xcode once from the Applications folder to initialize it.)

**Comment by myeomans** [comment]
Yes, my Xcode is up-to-date. The tests at the link flag a few errors for both SpaCy 1.9 (which works with spacyr) and SpaCy 2.0.3 (which does not), so who knows.

**Comment by chmue** [comment]
Sorry for jumping in on this but do you have a very advanced setup using the [Intel Math Kernel Library](https://software.intel.com/en-us/mkl) @myeomans ?

> "Intel MKL FATAL ERROR: Cannot load libmkl_intel_thread.dylib"

This message tells me that either R, reticulate, or spacy was compiled against the MKL but one of the libraries from the MKL you (?) compiled against is not present on your systema anymore. I am asking because if I interpret this error correctly, it happens at a much lower level than spacyr. I suspect that the problem is signalled from python and it might be that `numpy`, which is pulled in as a dependency by spacy, was built against the MKL. The `sessisonInfo()` above shows that R was not compiled against the MKL (the lines after the first empty line) so this should be fine.

**Comment by kbenoit** [comment]
@chmue never apologise for jumping in with solutions or ideas! Thanks!

**Comment by myeomans** [comment]
Yeah, it's a great suggestion thanks! My setup is quite vanilla, though, only the default installations...  Don't want to send you on a wild goose chase, though, if the issue is deeper than spacyr.

**Comment by chmue** [comment]
It might be that you just need to reinstall spacy and, importantly, its dependencies if you did not already do that. If what I suspect is true, you might have simply installed a strange compiled version of spacy or one of its dependencies, maybe due to a bug in pip/ananconda/wherever you get your packages from.

You can find the up-to-date list of spacy dependencies [here](https://github.com/explosion/spaCy/blob/master/requirements.txt). You see that the third line is numpy which links against BLAS/LAPACK.

**Comment by myeomans** [comment]
Update: fixed in SpaCy 2.0.5! Not sure what the solution was, but there you go.

**Comment by chmue** [comment]
Great to hear that it works now. The fact that it was resolved by a version bump/recompilation of spacy by your packager makes it likely that it was a transient problem in their pipeline.
